coding_systems/dmd/tests/test_coding_system.py

```diff
@@ -25,3 +25,87 @@ def test_code_to_term(dmd_data, coding_system):
         "10525011000001107": "Adrenaline (base) 220micrograms/dose inhaler refill (VMP)",
         "99999": "Unknown",
     }
+
+
+def test_ancestor_relationships(dmd_data, coding_system):
+    """
+    Test that given each of a code of type [AMP, VMP, VTM]
+    the ancestor relationships all the way up to the ultimate
+    ancestor VTM(s) are returned
+    """
+
+    # "Salbutamol" VTM has no ancestors
+    salbutamol_vtm = "777483005"
+    expected_relationships = set()
+
+    assert (
+        coding_system.ancestor_relationships([salbutamol_vtm]) == expected_relationships
+    )
+
+    # 35936411000001109:"Salbutamol 100micrograms/dose breath actuated inhaler"
+    # and 39112711000001103:"Salbutamol 100micrograms/dose breath actuated inhaler CFC free"
+    # share a common ancestor of 777483005:"Salbutamol"
+
+    salbutamol_vmps = ["35936411000001109", "39112711000001103"]
+    expected_relationships |= {(salbutamol_vtm, vmp) for vmp in salbutamol_vmps}
+
+    assert (
+        coding_system.ancestor_relationships(salbutamol_vmps) == expected_relationships
+    )
+
+    # "Adrenaline" VTM has no ancestors
+    adrenaline_vtm = "65502005"
+
+    # "Adrenaline (base) 220micrograms/dose inhaler" has an ancestor of "Adrenaline" VTM
+    # and no relation to Salbutamol
+    adrenaline_vmps = ["10514511000001106"]
+
+    assert coding_system.ancestor_relationships(salbutamol_vmps + adrenaline_vmps) == (
+        expected_relationships | {(adrenaline_vtm, vmp) for vmp in adrenaline_vmps}
+    )
+
+    # 3293111000001105:"Aerolin 100micrograms/dose Autohaler" has an ancestor of VMP 35936411000001109
+    # and, in turn, VTM 777483005
+    salbutamol_amps = ["3293111000001105"]
+    expected_relationships |= {("35936411000001109", "3293111000001105")}
+    assert (
+        coding_system.ancestor_relationships(salbutamol_vmps + salbutamol_amps)
+        == expected_relationships
+    )
+
```

Comment:
Should we add a test for a VMP/AMP without a VTM?

Reply:
Yes good point. I'm not sure I put one of those into the test data :facepalm:

Reply:
Bit of a slog, but I've added another commit for this